homeassistant/loader.py

```diff
@@ -1166,14 +1164,18 @@ async def async_get_platforms(
 
         return platforms
 
-    def _get_platform_cached_or_raise(self, full_name: str) -> ModuleType | None:
+    def _get_platform_cached_or_raise(self, platform_name: str) -> ModuleType | None:
         """Return a platform for an integration from cache."""
+        full_name = f"{self.domain}.{platform_name}"
         if full_name in self._cache:
             # the cache is either a ModuleType or a ComponentProtocol
             # but we only care about the ModuleType here
             return self._cache[full_name]  # type: ignore[return-value]
         if full_name in self._missing_platforms_cache:
-            raise self._missing_platforms_cache[full_name]
```

Comment:
It was the re-raise here that triggered the leak. By making a new exception it doesn't leak

Reply:
It didnt work to rase x from y?

Reply:
Nope, any time it was in the chain it leaked